src/prng.rs

```diff
@@ -61,74 +119,42 @@ pub enum PrngError {
 /// This type implements an iterator that generates a pseudorandom sequence of field elements. The
 /// sequence is derived from the key stream of AES-128 in CTR mode with a random IV.
 #[derive(Debug)]
-pub(crate) struct Prng<F: FieldElement> {
+pub(crate) struct Prng<F: FieldElement, X: StreamCipher> {
     phantom: PhantomData<F>,
-    cipher: Aes128Ctr,
-    length: Option<usize>,
+    key_stream: X,
     buffer: Vec<u8>,
     buffer_index: usize,
     output_written: usize,
 }
 
-impl<F: FieldElement> Prng<F> {
+impl<F: FieldElement, X: StreamCipher> Prng<F, X> {
     /// Generates a seed and constructs an iterator over an infinite sequence of pseudorandom field
     /// elements.
-    pub(crate) fn new() -> Result<Self, getrandom::Error> {
-        let mut seed = [0; SEED_LENGTH];
+    pub(crate) fn new() -> Result<Self, PrngError> {
+        let mut seed = vec![0; X::STREAM_CIPHER_SEED_SIZE];
```

Comment:
nit: make this a byte slice on the stack instead of a heap allocation that never escapes this function.